src/Npgsql/NpgsqlStatement.cs

```diff
@@ -106,7 +106,11 @@ public List<NpgsqlParameter> InputParameters
         /// <summary>
         /// Whether this statement has already been prepared (including automatic preparation).
         /// </summary>
-        internal bool IsPrepared => PreparedStatement?.IsPrepared == true;
+        internal bool IsPrepared([NotNullWhen(true)] out PreparedStatement? preparedStatement)
+        {
+            preparedStatement = PreparedStatement;
+            return PreparedStatement?.IsPrepared == true;
```

Comment:
This looks like it's still accessing PreparedStatement twice (that property has an actual getter):
```suggestion
            return preparedStatement?.IsPrepared == true;
```

Reply:
Oh come on 😢

Reply:
You don't like it? :) Why not?

Reply:
Not that - just annoyed I missed that.